databricks_cli/configure/provider.py

```diff
@@ -185,20 +185,41 @@ def get_config(self):
 
 
 class DefaultConfigProvider(DatabricksConfigProvider):
-    """Prefers environment variables, and then the default profile."""
+    """Look for credentials in a chain of default locations."""
     def __init__(self):
-        self.env_provider = EnvironmentVariableConfigProvider()
-        self.default_profile_provider = ProfileConfigProvider()
+        self._providers = (
+            SparkTaskContextConfigProvider(),
+            EnvironmentVariableConfigProvider(),
+            ProfileConfigProvider()
+        )
 
     def get_config(self):
-        env_config = self.env_provider.get_config()
-        if env_config:
-            return env_config
+        for provider in self._providers:
+            config = provider.get_config()
+            if config is not None and config.is_valid:
```

Comment:
Why is the `config.is_valid` piece necessary here?

Reply:
It's not technically necessary because the providers have this test inside and return None if not valid. 
But I think it's better to have this test here so a)  we make sure this is consistent across all (possible future) providers and b) it makes it clear that the intended behavior is to go through the chain until valid credentials are found.. 

I can also remove the test from the individual providers but I did not want to make more changes than necessary.